app/js/ui.js

```diff
@@ -1,29 +1,64 @@
 import 'whatwg-fetch';
 import React from 'react';
+import { autobind } from 'core-decorators';
+import classNames from 'classnames';
+import { App } from 'spak';
 
-class App extends React.Component {
+class AppVC extends React.Component {
     constructor() {
         super();
         this.state = {
-            styles: ''
+            styles: '',
+            hasLoginForm: false
         };
+        this._off = App.events.on("ui.requestLoginForm", this._showForm);
+    }
+
+    componentWillUnmount() {
+        this._off();
     }
 
     render() {
+        var loginForm = this.state.hasLoginForm ? this._loginForm : null;
+
         return (
-            <main>
-                <h2>Hey there!</h2>
-                <button onClick={() => this._handleFetchClick()}>Fetch Styles</button>
-                { this.state.styles }
-                <LoginForm />
+            <main className={classNames('appvc', this._classes)}>
+                <div className="appvc__loader progress">
+                    <div className="indeterminate"></div>
+                </div>
+                { loginForm }
             </main>
         );
     }
 
+    get _loginForm() {
+        return (
+            <div>
+                <LoginForm />
+                <h4>Ping api</h4>
+                <button className="btn" onClick={this._handleFetchClick}>Fetch Styles</button>
+                { this.state.styles }
+            </div>
+        );
+    }
+
+    get _classes() {
```

Comment:
`classNames` fn can be provided with a list of classes to apply to your element. If you provide an object, the keys are the class you want to show, and the value is a boolean. If true, it will place the class on the element, if false it won't. So in this scenario I wont the `is-loading` class to be applied when we are not showing the login form.

To keep it clean I put this in a `getter` fn so we can just access it like a property.